### ISSUE TITLE
Problem when trying to upgrade a version of Leo in an existing VE to Leo 6.0 beta 1

### ISSUE THREAD

**Issue by ranvik14** [leo-editor/leo-editor#1217]
When I tried to install Leo 6.0 beta 1 into an **existing*  virtual environment I ran into the following problem, which resulted in a state, where Leo was no longer usable.

I'm attaching two logs. - The first one shows some very basic information about the state of the VE before the installation - and - the second one shows the information reported by pip.

<Log - Part #1 >

(VE4Leo) PS C:\Users\Viktor\pyve\VE4Leo> pip show leo                                                                   Name: leo
Version: 6.0-devel
Summary: An IDE, PIM and Outliner
Home-page: http://leoeditor.com
Author: Edward K. Ream
Author-email: edreamleo@gmail.com
License: MIT License
Location: c:\users\viktor\pyve\ve4leo\lib\site-packages
Requires: nbformat, flexx, pyflakes, pylint, future, sphinx, six, PyQt5, docutils
Required-by:
(VE4Leo) PS C:\Users\Viktor\pyve\VE4Leo>  

</Log - Part #1>

<Log - Part #2>

(VE4Leo) PS C:\Users\Viktor\pyve\VE4Leo> pip install leo==6.0b1                                                         Collecting leo==6.0b1
  Downloading https://files.pythonhosted.org/packages/70/e5/3421fc20192f03c8378784f24dbbad3555966fdd4708b631023604cebe9d/leo-6.0b1-py2.py3-none-any.whl (9.4MB)
     |████████████████████████████████| 9.4MB 1.3MB/s
Requirement already satisfied: PyQt5; python_version >= "3.0" in c:\users\viktor\pyve\ve4leo\lib\site-packages (from leo==6.0b1) (5.11.3)
Requirement already satisfied: pyflakes in c:\users\viktor\pyve\ve4leo\lib\site-packages (from leo==6.0b1) (2.0.0)
Requirement already satisfied: docutils in c:\users\viktor\pyve\ve4leo\lib\site-packages (from leo==6.0b1) (0.14)
Requirement already satisfied: nbformat in c:\users\viktor\pyve\ve4leo\lib\site-packages (from leo==6.0b1) (4.4.0)
Requirement already satisfied: sphinx in c:\users\viktor\pyve\ve4leo\lib\site-packages (from leo==6.0b1) (1.8.3)
Requirement already satisfied: future in c:\users\viktor\pyve\ve4leo\lib\site-packages (from leo==6.0b1) (0.17.1)
Requirement already satisfied: pylint in c:\users\viktor\pyve\ve4leo\lib\site-packages (from leo==6.0b1) (2.2.2)
Requirement already satisfied: flexx; python_version >= "3.5" in c:\users\viktor\pyve\ve4leo\lib\site-packages (from leo==6.0b1) (0.8.0)
Collecting PyQtWebEngine (from leo==6.0b1)
  Downloading https://files.pythonhosted.org/packages/20/0f/d0d7b7e33bb3dc5411013886af43e8f922c2d8672d8173b6a5419e91ac56/PyQtWebEngine-5.12.1-5.12.3-cp35.cp36.cp37.cp38-none-win_amd64.whl (49.0MB)
     |████████████████████████████████| 49.0MB 595kB/s
Collecting meta (from leo==6.0b1)
  Downloading https://files.pythonhosted.org/packages/78/4e/bf6414dd628664eee7d03831f36fe9ffd43708c36c8bd43ae56ebe7cd69d/meta-1.0.2.tar.gz (49kB)
     |████████████████████████████████| 51kB 409kB/s
Requirement already satisfied: six in c:\users\viktor\pyve\ve4leo\lib\site-packages (from leo==6.0b1) (1.12.0)
Requirement already satisfied: PyQt5_sip<4.20,>=4.19.11 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from PyQt5; python_version >= "3.0"->leo==6.0b1) (4.19.13)
Requirement already satisfied: jupyter-core in c:\users\viktor\pyve\ve4leo\lib\site-packages (from nbformat->leo==6.0b1) (4.4.0)
Requirement already satisfied: jsonschema!=2.5.0,>=2.4 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from nbformat->leo==6.0b1) (2.6.0)
Requirement already satisfied: traitlets>=4.1 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from nbformat->leo==6.0b1) (4.3.2)
Requirement already satisfied: ipython-genutils in c:\users\viktor\pyve\ve4leo\lib\site-packages (from nbformat->leo==6.0b1) (0.2.0)
Requirement already satisfied: sphinxcontrib-websupport in c:\users\viktor\pyve\ve4leo\lib\site-packages (from sphinx->leo==6.0b1) (1.1.0)
Requirement already satisfied: colorama>=0.3.5; sys_platform == "win32" in c:\users\viktor\pyve\ve4leo\lib\site-packages (from sphinx->leo==6.0b1) (0.4.1)
Requirement already satisfied: requests>=2.0.0 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from sphinx->leo==6.0b1) (2.21.0)
Requirement already satisfied: babel!=2.0,>=1.3 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from sphinx->leo==6.0b1) (2.6.0)
Requirement already satisfied: snowballstemmer>=1.1 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from sphinx->leo==6.0b1) (1.2.1)
Requirement already satisfied: alabaster<0.8,>=0.7 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from sphinx->leo==6.0b1) (0.7.12)
Requirement already satisfied: imagesize in c:\users\viktor\pyve\ve4leo\lib\site-packages (from sphinx->leo==6.0b1) (1.1.0)
Requirement already satisfied: packaging in c:\users\viktor\pyve\ve4leo\lib\site-packages (from sphinx->leo==6.0b1) (18.0)
Requirement already satisfied: Jinja2>=2.3 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from sphinx->leo==6.0b1) (2.10)
Requirement already satisfied: setuptools in c:\users\viktor\pyve\ve4leo\lib\site-packages (from sphinx->leo==6.0b1) (39.0.1)
Requirement already satisfied: Pygments>=2.0 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from sphinx->leo==6.0b1) (2.3.1)
Requirement already satisfied: astroid>=2.0.0 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from pylint->leo==6.0b1) (2.1.0)
Requirement already satisfied: isort>=4.2.5 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from pylint->leo==6.0b1) (4.3.4)
Requirement already satisfied: mccabe in c:\users\viktor\pyve\ve4leo\lib\site-packages (from pylint->leo==6.0b1) (0.6.1)
Requirement already satisfied: dialite>=0.5.2 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from flexx; python_version >= "3.5"->leo==6.0b1) (0.5.2)
Requirement already satisfied: webruntime>=0.5.6 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from flexx; python_version >= "3.5"->leo==6.0b1) (0.5.6)
Requirement already satisfied: tornado in c:\users\viktor\pyve\ve4leo\lib\site-packages (from flexx; python_version >= "3.5"->leo==6.0b1) (6.0.2)
Requirement already satisfied: pscript>=0.7.1 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from flexx; python_version >= "3.5"->leo==6.0b1) (0.7.1)
Requirement already satisfied: decorator in c:\users\viktor\pyve\ve4leo\lib\site-packages (from traitlets>=4.1->nbformat->leo==6.0b1) (4.3.0)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from requests>=2.0.0->sphinx->leo==6.0b1) (3.0.4)
Requirement already satisfied: certifi>=2017.4.17 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from requests>=2.0.0->sphinx->leo==6.0b1) (2018.11.29)
Requirement already satisfied: idna<2.9,>=2.5 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from requests>=2.0.0->sphinx->leo==6.0b1) (2.8)
Requirement already satisfied: urllib3<1.25,>=1.21.1 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from requests>=2.0.0->sphinx->leo==6.0b1) (1.24.1)
Requirement already satisfied: pytz>=0a in c:\users\viktor\pyve\ve4leo\lib\site-packages (from babel!=2.0,>=1.3->sphinx->leo==6.0b1) (2018.7)
Requirement already satisfied: pyparsing>=2.0.2 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from packaging->sphinx->leo==6.0b1) (2.3.0)
Requirement already satisfied: MarkupSafe>=0.23 in c:\users\viktor\pyve\ve4leo\lib\site-packages (from Jinja2>=2.3->sphinx->leo==6.0b1) (1.1.0)
Requirement already satisfied: lazy-object-proxy in c:\users\viktor\pyve\ve4leo\lib\site-packages (from astroid>=2.0.0->pylint->leo==6.0b1) (1.3.1)
Requirement already satisfied: wrapt in c:\users\viktor\pyve\ve4leo\lib\site-packages (from astroid>=2.0.0->pylint->leo==6.0b1) (1.10.11)
Requirement already satisfied: typed-ast; python_version < "3.7" and implementation_name == "cpython" in c:\users\viktor\pyve\ve4leo\lib\site-packages (from astroid>=2.0.0->pylint->leo==6.0b1) (1.1.1)
Building wheels for collected packages: meta
  Building wheel for meta (setup.py) ... done
  Stored in directory: C:\Users\Viktor\AppData\Local\pip\Cache\wheels\c3\ae\57\3ab58d0d76a5828f892f3099bb4b55264395adde36c4f3e790
Successfully built meta
ERROR: pyqtwebengine 5.12.1 has requirement PyQt5>=5.12, but you'll have pyqt5 5.11.3 which is incompatible.
Installing collected packages: PyQtWebEngine, meta, leo
  Found existing installation: leo 6.0-devel
    Uninstalling leo-6.0-devel:
      Successfully uninstalled leo-6.0-devel
Successfully installed PyQtWebEngine-5.12.1 leo-6.0b1 meta-1.0.2
(VE4Leo) PS C:\Users\Viktor\pyve\VE4Leo>  

</Log - Part #2>

Please let me know if I should provide additional information. - I did not further modify this VE after the problem occured ...

**Comment by maphew** [comment]
Replicated on Win10, upgrading from leo-5.8.1b1.dev2018 to 6.0b1.dev66. Relevant part seems to be:
`pyqtwebengine 5.12.1 has requirement PyQt5>=5.12, but you'll have pyqt5 5.9.2 which is incompatible.`

Digging in...

**Comment by maphew** [comment]
Domino effect from the fix in #1202, naming `pyqtwebengine` as a dependency.

> PyQtWebEngine is a set of Python bindings for The Qt Company's Qt WebEngine libraries. The bindings sit on top of PyQt5 and are implemented as a set of three modules. Prior to v5.12 these bindings were part of PyQt itself. 
-- https://www.riverbankcomputing.com/software/pyqtwebengine/intro

So we need to require PyQt 5.12+ also. 
Done in commit 679fc60. tested and working on Win10 Surface x64.

To fix your VE `git pull` current devel branch and `pip install -e path/to/leo-editor`, or run `pip install --upgrade pyqt`. PyQt and PyQt-sip packages will be upgraded (~50mb download) and then Leo should "just work".

**Comment by edreamleo** [comment]
On Wed, Jun 26, 2019 at 1:36 AM matt wilkie <notifications@github.com>
wrote:

> Closed #1217 <https://github.com/leo-editor/leo-editor/issues/1217>.
>

Super.  Once again, thanks for this work.  And thanks again for the careful
testing in the original report.

Edward

**Comment by ranvik14** [comment]
Hello Matt,

Am Mi., 26. Juni 2019 um 08:36 Uhr schrieb matt wilkie <
notifications@github.com>:

> Domino effect from the fix in #1202
> <https://github.com/leo-editor/leo-editor/issues/1202>, naming
> pyqtwebengine as a dependency.
>
> PyQtWebEngine is a set of Python bindings for The Qt Company's Qt
> WebEngine libraries. The bindings sit on top of PyQt5 and are implemented
> as a set of three modules. Prior to v5.12 these bindings were part of PyQt
> itself.
> -- https://www.riverbankcomputing.com/software/pyqtwebengine/intro
>
> So we need to require PyQt 5.12+ also.
> Done in commit 679fc60
> <https://github.com/leo-editor/leo-editor/commit/679fc6047c7888993e6e25eb61a7508444a5126c>.
> tested and working on Win10 Surface x64.
>
I can confirm that I'm now able to upgrade an existing 5.9 VE to the latest
6.0-b2-devel VE.

However when I start this instance, and - **only this instance** - it is
not really usable anymore an. - Here's the manually created log, since
copy/ paste is not working anymore :-(

Leo Log Window
Leo 6.0-b2-devel
Python 3.6.5, PyQt version 5.12.3
Windows 10 AMD64 (build 10.0.18362) SP0
current dir: C:/Users/Viktor/pyve/VE4ILeo
load dir: C:/users/viktor/pyve/VE4ILeo/lib/site-packages/leo/core
global config dir: None
home dir: C:/Users/Viktor
reading settings in C:/Users/Viktor/.leo/myLeoSettings.leo
reading settings in C:/Users/Viktor/Worklogs/WL2019.leo
may be open in another Leo:
C:/Users/Viktor/Worklogs/WL2019.leo
read outline in 2.67 seconds
No @menu setting found
reading settings in C:/Users/Viktor/Worklogs/WL2018.leo

 Please let me know if you need further details and/or if I should create a
separate issue on GitHub ...

With kind regards,

Viktor

**Comment by maphew** [comment]
I would let this sleeping dog lie unless someone else runs into it too. You have other Leo VE's that work, so your projects are not blocked (right?). At any rate, at present I've not idea what else might be wrong here.

**Comment by edreamleo** [comment]
On Tue, Jul 9, 2019 at 12:46 AM matt wilkie <notifications@github.com>
wrote:

> I would let this sleeping dog lie unless someone else runs into it too.
> You have other Leo VE's that work, so your projects are not blocked
> (right?). At any rate, at present I've not idea what else might be wrong
> here.
>

I agree.  This looks like an obscure installation issue.

Edward
